app/bundles/LeadBundle/Form/Type/AddToCompanyActionType.php

```diff
@@ -0,0 +1,95 @@
+<?php
+/**
+ * @package     Mautic
+ * @copyright   2014 Mautic Contributors. All rights reserved.
+ * @author      Mautic
+ * @link        http://mautic.org
+ * @license     GNU/GPLv3 http://www.gnu.org/licenses/gpl-3.0.html
+ */
+
+namespace Mautic\LeadBundle\Form\Type;
+
+use Mautic\CoreBundle\Factory\MauticFactory;
+use Symfony\Component\Form\AbstractType;
+use Symfony\Component\Form\FormBuilderInterface;
+use Symfony\Component\Validator\Constraints\NotBlank;
+use Symfony\Component\OptionsResolver\OptionsResolverInterface;
+use Doctrine\ORM\EntityRepository;
+
+/**
+ * Class FormSubmitActionAddUtmTagType
+ *
+ * @package Mautic\LeadBundle\Form\Type
+ */
+class AddToCompanyActionType extends AbstractType
+{
+    private $factory;
+
+    /**
+     * @param MauticFactory $factory
+     */
+    public function __construct(MauticFactory $factory)
```

Comment:
Let's not use MauticFactory in new code as it's deprecated - pass in the `router` service instead